test/parallel_api/iterator/zip_iterator_merge.pass.cpp

```diff
@@ -0,0 +1,137 @@
+// -*- C++ -*-
+//===-- zip_iterator_merge.pass.cpp ---------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+#include "support/utils.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+
+#if !defined(_PSTL_TEST_MERGE)
```

Comment:
Why we need this code in this file and in some another new files too?

Reply:
It is a good question. I have just split up what was already in `zip_iterator.pass.cpp` into each new file. I agree if these are not needed, then we should just remove them.

Do you know who may have a good idea of why these were initially added?

Reply:
My understanding is that these are used for debugging purposes to isolate specific aspects of a large test.   Since that is effectively what this PR is doing, i don't think they are necessary anymore.

Reply:
Thanks for the clarification. I agree these guards are no longer necessary and have removed them.